internal/controller/platform/dbMigratorJob.go

```diff
@@ -0,0 +1,222 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package platform
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"strings"
+	"time"
+
+	operatorapi "github.com/apache/incubator-kie-kogito-serverless-operator/api/v1alpha08"
+	"github.com/apache/incubator-kie-kogito-serverless-operator/container-builder/client"
+	"github.com/apache/incubator-kie-kogito-serverless-operator/internal/controller/platform/services"
+
+	"github.com/apache/incubator-kie-kogito-serverless-operator/log"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	jobName                  = "kogito-db-migrator-job"
+	dbMigrationContainerName = "db-migration-container"
+	dbMigratorToolImage      = "quay.io/rhkp/incubator-kie-kogito-service-db-migration-postgresql:latest"
+	dbMigrationCmd           = "./migration.sh"
+)
+
+type DBMigratorJob struct {
+	migrateDbDataindex                   bool
+	quarkusDatasourceDataindexJdbcUrl    string
+	quarkusDatasourceDataindexUsername   string
+	quarkusDatasourceDataindexPassword   string
+	quarkusFlywayDataindexSchemas        string
+	migrateDbJobsservice                 bool
+	quarkusDatasourceJobsserviceJdbcUrl  string
+	quarkusDatasourceJobsserviceUsername string
+	quarkusDatasourceJobsservicePassword string
+	quarkusFlywayJobsserviceSchemas      string
+}
```

Comment:
There are many repetitions in this data structure. Rather than doing this, you can do the following:

```go
type QuarkusDataSource struct {
    JdbcUrl string
    Username string
    Password string
    Schema string
}

type DBMigratorJob struct {
     DataIndexMigrate bool
     DataIndexDataSource QuarkusDataSource
     JobsService bool
     JobsServiceDataSource QuarkusDataSource
}
```